src/poetry/utils/env/env_manager.py

```diff
@@ -405,10 +405,34 @@ def remove(self, python: str) -> Env:
                     del envs[base_env_name]
                     envs_file.write(envs)
 
-        self.remove_venv(venv_path)
+        self.remove_venv_dir(venv_path)
 
         return VirtualEnv(venv_path, venv_path)
 
+    def remove_venv(self, venv: VirtualEnv) -> Env:
+        venv_path = self._poetry.config.virtualenvs_path
+        cwd = self._poetry.file.path.parent
+        envs_file = TOMLFile(venv_path / self.ENVS_FILE)
+        base_env_name = self.generate_env_name(self._poetry.package.name, str(cwd))
+
+        env_path = venv.path
+        _, minor, _ = venv.get_version_info()
+
+        name = f"{base_env_name}-py{minor}"
+
+        if not env_path.exists():
+            raise ValueError(f'<warning>Environment "{name}" does not exist.</warning>')
+
+        if envs_file.exists():
+            envs = envs_file.read()
+            if base_env_name in envs:
+                del envs[base_env_name]
+                envs_file.write(envs)
```

Comment:
It may not make a difference for `remove --all` but considering that this method removes exactly one venv, we should only remove the entry if the Python version matches the version of the venv we are removing.

Reply:
Indeed, this may no make a difference. The check is only there if there is an inconsistency between the TOML env file and the virtual env folders.